transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java

```diff
@@ -172,7 +172,7 @@ private void incrementPendingOutboundBytes(long size, boolean invokeLater) {
         }
 
         long newWriteBufferSize = TOTAL_PENDING_SIZE_UPDATER.addAndGet(this, size);
-        if (newWriteBufferSize > channel.config().getWriteBufferHighWaterMark()) {
+        if (newWriteBufferSize > channel.config().getWriteBufferHighWaterMark() && isWritable()) {
```

Comment:
is isWritable a non volatile boolean check? if yes, it should go first.

Reply:
(cheaper expressions should come first)

Reply:
``newWriteBufferSize > channel.config().getWriteBufferHighWaterMark()`` is a statement that must be executed, but ``isWritable()`` is not, So I think ``newWriteBufferSize > channel.config().getWriteBufferHighWaterMark()`` should be the first

Reply:
both are volatile so I think it not really matters.

Reply:
Can this PR be merged? @normanmaurer

Reply:
@brucelwl I want to at least have one more person to review it.

Reply:
I'm travelling atm and will come back on late Friday: I would like to review this so please, if there is no hurry, wait for me so I can play a bit with it :)

Reply:
@franz1981 sounds like a plan... safe travels!